### ISSUE TITLE
RS1024 False-positive for Enumerable.Contains method

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#4469]
### Analyzer

**Diagnostic ID**: RS1024 `Compare symbols correctly`

### Analyzer source

````
    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.1" PrivateAssets="all" />
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.8.0" PrivateAssets="all" />
````

### Describe the bug

````
// using System.Linq;
IEnumerable<ISymbol> symbols = Enumerable.Empty<ISymbol>();
ISymbol symbol = default!;
symbols.Contains(symbol, SymbolEqualityComparer.Default); // RS1024
````

### Steps to reproduce

You can check the code from my project

1. Clone https://github.com/meziantou/Meziantou.Analyzer/
2. Checkout f93959b7efc062118db672f4acab177bf1deb846
3. Open the solution in Visual Studio

### Expected behavior

No diagnostic is reported as I use `SymbolEqualityComparer.Default`

### Actual behavior

RS1024 is reported

**Comment by Evangelink** [comment]
I can confirm the behavior. I don't know if there is a regression in Roslyn or if that was just a case we missed.

**Comment by Dimension4** [comment]
Does this fix also propagate to the `Microsoft.CodeAnalysis.Analyzers` package? That package hasn't been updated for 2 months on nuget so it does not include this fix.

**Comment by Evangelink** [comment]
I don't think it will, @mavasani ?
